IPython/kernel/zmq/kernelbase.py

```diff
@@ -699,3 +701,14 @@ def _at_shutdown(self):
             self.session.send(self.iopub_socket, self._shutdown_message, ident=self._topic('shutdown'))
             self.log.debug("%s", self._shutdown_message)
         [ s.flush(zmq.POLLOUT) for s in self.shell_streams ]
+
+    def _flush_execute_output(self):
```

Comment:
It is specific to execute command because it sleeps for `_execute_sleep` delay, that's why I added 'execute' to the name. But there are other places where this function can be useful - should they also sleep? If so, I think it could make sense to rename `_execute_sleep`.